### PR TITLE
refactor  tskv : Error passed up in async function

### PR DESCRIPTION
**Pull request by Babanmei**
Errors inside the [run_wal_job,run_flush_job,run_compact_job,run_summary_job] functions are passed up

避免以上四个函数中的错误在内部消耗而外部不可知而向上传递,并将方法改为异步函数形式,另外项目Error设计不是很优雅

### CODE REVIEW

**Comment by yuandongs** [comment]
Thank You

**Comment by yuandongs** [comment]
@roseboy-liu please make a review

File: tskv/src/kvcore.rs [hunk 1]
```diff
@@ -475,3 +379,86 @@ impl TsKv {
         Ok(None)
     }
 }
+
+
+async fn run_wal_job(wal_opt: Arc<WalConfig>, mut receiver: UnboundedReceiver<WalTask>) -> Result<()> {
```
Comment (roseboy-liu) [inline tskv/src/kvcore.rs:384]: If there is no refactoring， in place to modify

Comment (Babanmei) [reply tskv/src/kvcore.rs:384]: change to function, remove &self

**Pull request closed without merging**